### ISSUE TITLE
prep_train.py

### ISSUE THREAD

**Issue by ansib1e** [avanetten/yoltv5#2]
For creating training datasets, is it possible to train the YOLTv5 model using data that was setup for the YOLOv5 format or must it go through the prep_train.py processing to be accepted for training by YOLTv5? Also just to confirm, but for creating training datasets using prep_train.py, are the only two inputs required the training image (.tif) and relavent geojson representing bounding boxes within this image? The main function () seems to be missing the same level of comments/clarification as the other functions in prep_train.py. 

P.S. Just curious, for the helper function in prep_train.py called "yolo_from_df", a dataframe of polygons is required as input which the comments said could be created using: DA_Dataset/gj_to_px_bboxes.ipynb. I cant seem to find that ipynb notebooko anywhere in this repo or online. Some directions on how to access this would be greatly appreciated so I can try creating training datasets to test out YOLTv5

Thanks!

**Comment by weypro** [comment]
HI. I'm trying to train the model by myself.  You can see [https://github.com/avanetten/yoltv4/blob/master/notebooks/train_test_pipeline.ipynb](https://github.com/avanetten/yoltv4/blob/master/notebooks/train_test_pipeline.ipynb)

**Comment by im-tanyasuri** [comment]
> For creating training datasets, is it possible to train the YOLTv5 model using data that was setup for the YOLOv5 format or must it go through the prep_train.py processing to be accepted for training by YOLTv5? Also just to confirm, but for creating training datasets using prep_train.py, are the only two inputs required the training image (.tif) and relavent geojson representing bounding boxes within this image? The main function () seems to be missing the same level of comments/clarification as the other functions in prep_train.py.
> 
> P.S. Just curious, for the helper function in prep_train.py called "yolo_from_df", a dataframe of polygons is required as input which the comments said could be created using: DA_Dataset/gj_to_px_bboxes.ipynb. I cant seem to find that ipynb notebooko anywhere in this repo or online. Some directions on how to access this would be greatly appreciated so I can try creating training datasets to test out YOLTv5
> 
> Thanks!

Hi,.. I also have the same question, what value does prep_train.py brings to the dataset,. Please let me know, if you have understood it.

**Comment by bling-bling-only** [comment]
hi，How do I apply the ordinary data set instead of the remote sensing data set  .I 'm trying to train my own data set and don 't know what to modify.

**Comment by ShenAoChen2001** [comment]
> > For creating training datasets, is it possible to train the YOLTv5 model using data that was setup for the YOLOv5 format or must it go through the prep_train.py processing to be accepted for training by YOLTv5? Also just to confirm, but for creating training datasets using prep_train.py, are the only two inputs required the training image (.tif) and relavent geojson representing bounding boxes within this image? The main function () seems to be missing the same level of comments/clarification as the other functions in prep_train.py.
> > P.S. Just curious, for the helper function in prep_train.py called "yolo_from_df", a dataframe of polygons is required as input which the comments said could be created using: DA_Dataset/gj_to_px_bboxes.ipynb. I cant seem to find that ipynb notebooko anywhere in this repo or online. Some directions on how to access this would be greatly appreciated so I can try creating training datasets to test out YOLTv5
> > Thanks!
> 
> Hi,.. I also have the same question, what value does prep_train.py brings to the dataset,. Please let me know, if you have understood it.

I also have the same problem,if you have understood it,Please teach me how to deal with it,thanks!!!!!!!!

**Comment by meisu1** [comment]
> hi，How do I apply the ordinary data set instead of the remote sensing data set .I 'm trying to train my own data set and don 't know what to modify.

hi，have you already done it ? Shall we have a chat？ please send me your contact details

**Comment by bling-bling-only** [comment]
> > hi，How do I apply the ordinary data set instead of the remote sensing data set .I 'm trying to train my own data set and don 't know what to modify.
> 
> hi，have you already done it ? Shall we have a chat？ please send me your contact details

No, recently busy in other areas, but still to try, if you have a good approach can communicate more

**Comment by ShenAoChen2001** [comment]
hello,do you have wechat？Cccsa2001 is my wechat account，you can add me



------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年4月10日(星期一) 下午4:34
收件人: ***@***.***&gt;; 
抄送: "(12N15L24) ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [avanetten/yoltv5] prep_train.py (Issue #2)





   
hi，How do I apply the ordinary data set instead of the remote sensing data set .I 'm trying to train my own data set and don 't know what to modify.
  
hi，have you already done it ? Shall we have a chat？ please send me your contact details
  
No, recently busy in other areas, but still to try, if you have a good approach can communicate more

**Comment by meisu1** [comment]
> > > hi，How do I apply the ordinary data set instead of the remote sensing data set .I 'm trying to train my own data set and don 't know what to modify.
> > 
> > 
> > hi，have you already done it ? Shall we have a chat？ please send me your contact details
> 
> No, recently busy in other areas, but still to try, if you have a good approach can communicate more

I've done modifying the parameters in the yaml file and there is no error utill the end of run. I use the  .pt file that I have trained for my datasets and it can detect target , but there is few detected target in the result file.
